src/store/userSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface User {
+  username: string;
+}
+
+export interface UserState {
+  userInfo: User | null;
+}
+
+export const initialState: UserState = {
+  userInfo: null,
+};
+
+const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    authUser(state: UserState, action: PayloadAction<User>) {
+      state.userInfo = action.payload;
```

Comment:
Is it advisable that `state` remains the same `state` and that you would only change its subfields?
We always made sure that the reducer creates an entirely new reference for `state` (`{...state, userInfo: action.payload}`)